### ISSUE TITLE
[bug] protocol message parser failed when running in multiple threads

### ISSUE THREAD

**Issue by EanCuznaivy** [AElfProject/AElf#155]
An exception (**sometimes**) threw when we run the test case _TxMemPool.IntergrationTest.StartMultiThread_:

`Google.Protobuf.InvalidProtocolBufferException : Protocol message contained an invalid tag (zero).`

Later we found the position causing this failure is a method of ChangesStore:

```C#
        public async Task<Change> GetChangeAsync(Hash pathHash)
        {
            var value = await _keyValueDatabase.GetAsync(pathHash.Value.ToBase64(), typeof(Change));
            return value == null ? null : Change.Parser.ParseFrom(value);
        }
```
It seems like something went wrong with the Parser of Change (deep research shows Protobuf failed to `ReadTag` of input data). But before long we found a interesting/weird/upset solution: by slightly modifying the implementation of AccountContextService.SetAccountContext:

From
```C#
await adp.GetDataProvider().SetAsync(GetKeyForIncrementId(), accountDataContext.IncrementId.ToBytes());
```
To
```C#
await adp.GetDataProvider().SetAsync(GetKeyForIncrementId(), new UInt64Value
            {
                Value = accountDataContext.IncrementId
            }.ToByteArray());
```

Basically we put a `ulong` type to a protobuf type `UInt64Value`, so at first we thought the reason cased aforementioned failure is the stuff we tried to set to database ain't a protobuf type. But during the attempt to confirm our conjecture, we found that the code actually does not reach the interaction of database, also the logic of Change is none of its business.

That's why I have to record this situation as an issue as well as seeking for help.
